api/src/main/java/org/apache/iceberg/Files.java

```diff
@@ -76,7 +75,7 @@ public PositionOutputStream create() {
     public PositionOutputStream createOrOverwrite() {
       if (file.exists()) {
         if (!file.delete()) {
-          throw new RuntimeIOException("Failed to delete: " + file);
+          throw new RuntimeIOException("Failed to delete: %s", file);
```

Comment:
I believe that this is technically more performant as it does not generate three strings like the previous version, which generated `"Failed to delete: "`, the string for `file` and then another string for the final result (without using StringBuilder).

This case should generate one less string at runtime: the `file` string and the templated `Failed to delete: $file`. Additionally, I believe the use of the format string is lazily evaluated (so the string won't be created if the exception is not thrown), but I might be mistaken on that part.

Probably the tldr is that it's better because the linting automation enforces it currently and thus it brings standardization to the code base.

Reply:
Yeah the main reason here is like Kyle said, 

t
The linter can't check a non effectively final format string. This is because the linter can't check at compile time whether or not the string is valid.

This is a case where we were basically calling format on the result of another format method with the exact same args. So the solution that makes the most sense is just eliminating the unneeded format call.

Reply:
Thanks for the clarification!

> Probably the best answer / tldr is that it's better because the linting automation enforces it currently and it brings standardization to the code base.

Make sense.

For rarely travelled code paths, I hope developers do not have to invest too much time tuning performance. Often compiler and GC already do a good enough job.

Reply:
Yeah I mostly made these changes because they register as a compiler error with the FormatString Specification. It doesn't let you do anything that could sneak by an invalid format.